src/snarky.d.ts

```diff
@@ -18,31 +18,124 @@ export {
+ * `Provable<T>` is the general circuit type interface in SnarkyJS. `Provable<T>` interface describes how a type `T` is made up of {@link Field} elements and "auxiliary" (non-provable) data.
  *
- * You will find `Provable<T>` as the required input type in a few places in SnarkyJS. One convenient way to create a `Provable<T>` is using `Struct`.
+ * `Provable<T>` is the required input type in a few places in SnarkyJS. One convenient way to create a `Provable<T>` is using `Struct`.
  * 
  * The properties and methods on the provable type exist in all base SnarkyJS types as well (aka. {@link Field}, {@link Bool}, etc.). In most cases, a zkApp developer does not need these functions to create Dapps.
  */
 declare interface Provable<T> {
-  toFields: (x: T) => Field[];
-  toAuxiliary: (x?: T) => any[];
-  fromFields: (x: Field[], aux: any[]) => T;
+  /**
+   * A function that takes `value`, an element of type `T`, as argument and returns an array of {@link Field} elements that made up the provable data of `value`.
+   * 
+   * @param value - the element of type `T` to generate the {@link Field} array from.
+   * 
+   * @return A {@link Field} array describing how this `T` element is made up of {@link Field} elements.
+   */
+  toFields: (value: T) => Field[];
+
+   /**
+   * A function that takes `value` (optional), an element of type `T`, as argument and returns an array of any type that made up the "auxilary" (non-provable) data of `value`.
+   * 
+   * @param value - the element of type `T` to generate the auxilary data array from, optional. If not provided, an empty array is returned.
+   * 
+   * @return An array of any type describing how this `T` element is made up of "auxilary" (non-provable) data.
+   */
+  toAuxiliary: (value?: T) => any[];
+
+  /**
+   * A function that returns an element of type `T` from the given provable and "auxilary" data.
+   * 
+   * **Important**: For any element of type `T`, this function is the reverse operation of calling {@link toFields} and {@link toAuxilary} methods on an element of type `T`.
+   * 
+   * @param fields - an array of {@link Field} elements describing the provable data of the new `T` element.
+   * @param aux - an array of any type describing the "auxilary" data of the new `T` element, optional.
+   * 
+   * @return An element of type `T` generated from the given provable and "auxilary" data.
+   */
+  fromFields: (fields: Field[], aux: any[]) => T;
+
+  /**
+   * Return the size of the `T` type in terms of {@link Field} type, as {@link Field} is the primitive type.
+   * 
+   * **Warning**: This function returns a `number`, so you cannot use it to prove something on chain. You can use it during debugging or to understand the memory complexity of some type.
+   * 
+   * **Beware**: Calling the statement `x.sizeInFields()` is equivalent to `x.toFields().length`.
+   * 
```

Comment:
This is not precise, the `Provable<T>` interface consists of static methods, and is needed internally if we _don't_ have an example instance `x`.

I would suggest to delete this sentence!

```suggestion
   * 
```